### DISCUSSION TITLE
情報交換

### DISCUSSION THREAD

**Discussion by akiray3** [akiray3/r-jp-doc#3] [Announcements]
Rのヘルプが使いにくい点
・英語で書いてある
・別ウィンドウが勝手に開いてしまって、consoleが隠れる
・スマホで見られない

**Comment by akiray3** [comment]
もっと詳しく知りたい。

![IMG_0024](https://github.com/akiray3/r-jp-doc/assets/129808898/67b3bd5e-71f2-44b3-b18d-98cf5e553c8a)
![IMG_0022](https://github.com/akiray3/r-jp-doc/assets/129808898/7ea7ec8f-6d36-4eb9-8358-41fe9657ab2b)
![IMG_0021](https://github.com/akiray3/r-jp-doc/assets/129808898/2c9d68ae-fdbb-4e5d-b09a-cd34e5a4a64d)

**Comment by aharu6** [comment]
1案目
![IMG_0026](https://github.com/akiray3/r-jp-doc/assets/85387680/8e16f86f-9b43-4d37-a056-d4742f6f7229)

２案目
![IMG_0028](https://github.com/akiray3/r-jp-doc/assets/85387680/fbf1b470-b635-44cf-b71a-66d18c171730)

関数のボタンをクリックしたら説明文が見れる。
検索したら該当関数が引っかかって表示。詳細ページに飛べる。
エラー検索は、ユーザーにエラーをコピペしてもらうつもりです。

**Reply by akiray3** [reply]
https://www.rdocumentation.org/

rdocumentationってサイトもあるんだけど、1案目とかのほうが使い勝手良さそう。

https://www.rdocumentation.org/packages/graphics/versions/3.6.2/topics/plot

これはplot関数のページ。
あ、アフィリエイト広告貼りたい笑

**Reply by aharu6** [reply]
では一案目で。こちらとしても作りやすいです。

広告、貼るのは賛成です。
作業は後で。

**Reply by akiray3** [reply]
ではこっちでDBの作成進めるので、ひとまずは画面の構築を自由に進めて貰えれば。
DBが成形できたら一気に機能付けてく感じで、そのときは手分けしてやっていこう。
全体的にまったり進行で。

**Comment by akiray3** [comment]
# データベースの進捗

JSで扱いやすいjson形式で、helpテキストのファイルを試作した（help_db_raw.json）。

1780行 x 4列のデータフレームに、ひとまずbase・stats・graphicsの関数のヘルプを全部記録した。
普通のやり方だと、Rの関数ヘルプは別窓で開いてしまい、consoleに表示されないし、オブジェクトに保存することもできなかった。

`x <- utils:::.getHelpFile(help(topic = 関数名))`

そこで、上記の開発者用のコードがブレークスルーとなった。
これを実行すると、markdown形式のhelpが格納される。

データフレームの4列は、pack（package; パッケージ）、func（funciton; 関数）、var（variable; 項目）、val（value; 値）。
現在varは”all_txt_tmp”という、markdown形式のhelpファイルをそのまま読み込んだもの、
今後、varは、各関数に対してArguments・Values・Examplesなど、ヘルプの章立てに沿って現在のテキストを分割格納したい。

<img width="578" alt="image" src="https://github.com/akiray3/r-jp-doc/assets/129808898/dd59f08e-cbde-4059-a582-658db003503e">

**Reply by akiray3** [reply]
現在のDBはさらに仕様を変えて、各列にはヘルプから取り出した文章をhtml形式で格納している。JSでうまく取り出せれば、楽に表示できる気がする。

**Reply by akiray3** [reply]
![image](https://github.com/akiray3/r-jp-doc/assets/129808898/1a5d0c5c-35cb-4202-ac49-8ba5b6540ce3)

現在のDBはこんな感じになった。各セルに保存されているのはhtml形式のヘルプ。
試験的に、いまのところpackageはbase, stats, graphics, psychの4つ。
関数の種類はぜんぶで2274種類。
確認用にexcelも作成した。

あとは各セルのテキスト部分をすべて日本語に翻訳すればおおむね完成。

**Comment by aharu6** [comment]
### 進捗
色々試して、base関数一覧分のボタンとそれぞれのリンクに飛べるページを作成。
思っていたより関数の分類が進まず、とりあえず全部並べるだけの状態。分類は出先で進めることにして、
ホームのページと、関数の紹介ページを一個作ることに進む。
キリが良いところで一度mainブランチへのmergeリクエストを出したいです

**Reply by akiray3** [reply]
リクエストはどんどん出してもらって大丈夫。スマホで承認できるから。
こちらは最近Onedriveの調子が悪いので、デスクトップとノートパソコンの作業同期するのに使ってる。

**Comment by akiray3** [comment]
@aharu6 

DBができつつあるけど、どんな形式だと扱いやすい？

現状のDBはこんな形で、

![image](https://github.com/akiray3/r-jp-doc/assets/129808898/5358dfbb-ef0c-4768-8b3e-06091b6e75a1)

たとえば1行目の関数"-"のArguments（引数）を取り出すと、

`<h3>Arguments</h3><table summary=\"R argblock\"><tr valign=\"top\"><td><code>x, y</code></td><td><p>numeric or complex vectors or objects which can becoerced to such, or other objects for which methods have been written.</p></td></tr></table>`

これが出てきます。htmlのコードがそのまま文字列として保存されているので、
うまく使えれば元々のヘルプの書式を流用できると思う。
これから私がやりたいのは、`<p>地の文</p>`だけを翻訳用のAPIで日本語にするって作業。

それにしてもこのページデザインすごく良いと思う！

https://github.com/akiray3/r-jp-doc/assets/129808898/870d301b-dab2-4478-a0f1-4757667caf8a

**Reply by akiray3** [reply]
ChatGPTにコードの解説をしてもらった。なるほど。

```
<!-- HTMLのバージョンを宣言（DOCTYPEはDOCTYPE HTMLを意味し、HTML5を使用することを宣言しています） -->
<!DOCTYPE html>
<head>
  <!-- ページの文字エンコーディングをUTF-8に設定（多言語対応） -->
  <meta charset="UTF-8" />
  <!-- ページのタイトルを設定 -->
  <title>home</title>
  <!-- CSSのリンクを設定（ページのデザインを制御） -->
  <link rel="stylesheet" href="style.css" type="text/css" />
  <!-- Slick（スライダーライブラリ）のテーマCSSをリンク -->
  <link
    rel="stylesheet"
    href="slick-1.8.1/slick/slick-theme.css"
    type="text/css"
  />
  <!-- Slickの基本CSSをリンク -->
  <link rel="stylesheet" href="slick-1.8.1/slick/slick.css" type="text/css" />
</head>
<body>
  <!-- フォームを作成（ユーザーが情報を入力し、送信するためのエリア） -->
  <!-- ここでは、ユーザーがキーワードを入力し、その結果をsearch.htmlにGETメソッドで送信するという設定になっています -->
  <form action="search.html" method="get">
    <input type="text" name="keyword" placeholder="キーワードを入力" />
    <!-- ボタンのidが"reserch"になっていますが、何もイベントが設定されていないので何も動作しないはずです -->
    <button id="reserch"></button>
  </form>

  <!-- sec_innerというクラスを持つdiv要素を開始（ページのセクションを定義） -->
  <!-- スタイルはインラインで定義されています（これは通常は推奨されません。通常は外部CSSファイルで定義します） -->
  <!-- 以下、base, graphics, psychの各セクションで同様のスタイリングが適用されています -->
  <div
    class="sec_inner"
    style="
      white-space: nrowrap;
      width: 100%;
      height: 250px;
      border-radius: 9.194px;
      stroke: rgb(36, 61, 37);
      stroke-miterlimit: 10;
      stroke-width: 1;
      border: 2px solid rgb(0, 0, 0);
      transform: translate(21.529px, 57.832px);
    "
  >
    <!-- baseセクションのタイトルとナビゲーション -->
    <div
      style="position: fixed; right: 0; top: 0; transform: translateX(-12px)"
    >
      <h1 style="font-size: 50px">base</h1>
      <!-- ナビゲーションリンク -->
      <nav>
        <a href="/infopage/base.html" style="padding: 15px">一覧を見る</a>
      </nav>
    </div>
    <!-- イメージとテキストを含むスライダーコンテンツ -->
    <!-- slick-sliderとborder-boxクラスを持つdiv要素に囲まれた複数のslick-list -->
    <!-- これらの各slick-listはリンクとイメージ（おそらくスライダーの各スライド）とテキストを含んでいます -->
    <!-- 以下、graphics, psychの各セクションでも同様の構造が見られます -->
    <div class="border-box" style="width: 80%; transform: translate(24px)">
      <div class="slick-slider">
        <div class="slick-list">
          <a href="/detail/base/apply.html">
            <img
              src="www/square.svg"
              alt=""
              style="height: 200px; width: 200px"
            />
          </a>
          <text>apply</text>
        </div>
        <!-- 他のslick-list要素は省略 -->
      </div>
    </div>
  </div>
  
  <!-- 他のsec_inner要素（graphics, psych）は省略 -->

  <!-- 必要なJavaScriptファイルをページに追加 -->
  <!-- jQuery（JavaScriptのライブラリ）、Slick（スライダーライブラリ）、および独自のJavaScriptファイル（script.js）をロードしています -->
  <script src="jquery.js"></script>
  <script src="slick-1.8.1/slick/slick.min.js"></script>
  <script src="script.js"></script>
</body>
```

**Reply by aharu6** [reply]
基本的には今の形式を想定してますが、引数ごとの列があったら組み込みやすいかなと想像してます。
dbのほうが早そうなので、基本そちらに合わせます。
画面の構成も修整したいので、このあと送ります。

slickslider気に入って頂けてよかったです。

**Reply by akiray3** [reply]
たしかに！
引数の種類は関数によってまちまちだけど、引数ごとにに細分化しておいたほうが取り出しやすいかもね。
というか、dbから1つの関数を取り出す時に、例えば引数が3種類あったら、3列に分けるようにしましょうか。
dbを作りつつ、JSでjsonを読み込むメソッドを作って渡しますね。

**Reply by aharu6** [reply]
![image](https://github.com/akiray3/r-jp-doc/assets/85387680/952544c3-f84f-4a8d-bbcd-8ac79c4fd8e7)

ありがとうございます

**Reply by akiray3** [reply]
なるほど！
それぞれの引数をラベルにして、独立したオブジェクトにするのだね。楽しそう！

**Comment by aharu6** [comment]
![image](https://github.com/akiray3/r-jp-doc/assets/85387680/50687847-1664-4db9-b98b-68a224e88803)

DS_store再び発生。コマンドで削除&commitに組み込まないようにする。

**Reply by akiray3** [reply]
また…！
思い当たるのは吉田のノートPCですね！
削除しときます！
